app/controllers/api/v1/admin/users_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Api::V1::Admin::UsersController < Api::V1::BaseController
+  before_action :authenticate_user!
+  before_action :authorize_user
+  before_action :load_user, except: [:index, :roles, :statuses]
+  after_action :verify_authorized
+
+  def index
+    @users = User.all
+    respond_with @users
+  end
+
+  def roles
+    respond_with User.roles.keys
+  end
+
+  def statuses
+    respond_with User.statuses.keys
+  end
+
+  def salary
```

Comment:
This controller is not about `salary` at all. This is a completely different resource.